src/libstd/collections/hash/raw/sse2.rs

```diff
@@ -0,0 +1,98 @@
+use super::bitmask::BitMask;
+use super::EMPTY;
+use core::mem;
+
+#[cfg(target_arch = "x86")]
+use core::arch::x86;
+#[cfg(target_arch = "x86_64")]
+use core::arch::x86_64 as x86;
+
+pub type BitMaskWord = u16;
+pub const BITMASK_SHIFT: u32 = 0;
+pub const BITMASK_MASK: u16 = 0xffff;
+
+/// Abstraction over a group of control bytes which can be scanned in
+/// parallel.
+///
+/// This implementation uses a 128-bit SSE value.
+pub struct Group(x86::__m128i);
+
+impl Group {
+    /// Number of bytes in the group.
+    pub const WIDTH: usize = mem::size_of::<Self>();
+
+    /// Returns a full group of empty bytes, suitable for use as the initial
+    /// value for an empty hash table.
+    ///
+    /// This is guaranteed to be aligned to the group size.
+    #[inline]
+    pub fn static_empty() -> &'static [u8] {
+        #[repr(C)]
+        struct Dummy {
+            _align: [x86::__m128i; 0],
+            bytes: [u8; Group::WIDTH],
+        };
+        const DUMMY: Dummy = Dummy {
+            _align: [],
+            bytes: [EMPTY; Group::WIDTH],
+        };
+        &DUMMY.bytes
+    }
+
+    /// Loads a group of bytes starting at the given address.
+    #[inline]
+    pub unsafe fn load(ptr: *const u8) -> Group {
+        Group(x86::_mm_loadu_si128(ptr as *const _))
+    }
+
+    /// Loads a group of bytes starting at the given address, which must be
+    /// aligned to `WIDTH`.
```

Comment:
It makes more sense to me to state that `ptr` must be aligned to `mem::align_of::<__m128>()`. It just so happens that `size_of::<__m128>() == align_of::<__m128>()` which is why this is true.

Reply:
This comment is intended for the user of the `Group` type. The `__m128i` is an target-specific  implementation detail.